### PR TITLE
AWS: fix inconsistency of S3 checksum feature names

### PR DESCRIPTION
**Pull request by jackye1995**

### CODE REVIEW

**Comment by jackye1995** [comment]
@arminnajafi @rajarshisarkar @singhpk234 @amogh-jahagirdar

File: aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java [hunk 1]
```diff
@@ -165,6 +165,12 @@
    */
   public static final String S3FILEIO_SESSION_TOKEN = "s3.session-token";
 
+  /**
+   * Enables eTag checks for S3 PUT and MULTIPART upload requests.
+   */
+  public static final String S3_CHECKSUM_ENABLED = "s3.checksum-enabled";
```
Comment (jackye1995) [inline aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java:171]: also moved this up to be together with other S3 properties.

File: aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java [hunk 2]
```diff
@@ -290,11 +296,11 @@ public AwsProperties(Map<String, String> properties) {
     Preconditions.checkArgument(s3FileIoAcl == null || !s3FileIoAcl.equals(ObjectCannedACL.UNKNOWN_TO_SDK_VERSION),
         "Cannot support S3 CannedACL " + aclType);
 
+    this.isS3ChecksumEnabled = PropertyUtil.propertyAsBoolean(properties, S3_CHECKSUM_ENABLED,
```
Comment (jackye1995) [inline aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java:299]: also moved this up to be together with other S3 properties.

File: aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java [hunk 3]
```diff
@@ -210,9 +216,8 @@
   public static final String CLIENT_ASSUME_ROLE_REGION = "client.assume-role.region";
 
   /**
-   * Enables eTag checks for S3 PUT and MULTIPART upload requests.
+   * @deprecated will be removed at 0.15.0, please use {@link #S3_CHECKSUM_ENABLED_DEFAULT} instead
    */
-  public static final String S3_CHECKSUM_ENABLED = "s3.checksum-enabled";
   public static final boolean CLIENT_ENABLE_ETAG_CHECK_DEFAULT = false;
```
Comment (singhpk234) [inline aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java:221]: [minor] we might need to add `@Deprecated` here as well.

Comment (jackye1995) [reply aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java:221]: good point, adding

**Comment by amogh-jahagirdar** [comment]
Looks good to me, thanks for the fix!

**Comment by jackye1995** [comment]
Thanks for the review! @amogh-jahagirdar @singhpk234

**Pull request merged**